src/main/java/com/flightstats/filesystem/FileSystem.java

```diff
@@ -20,6 +20,8 @@
 
     void saveContent(String content, Path fileName, String contentType);
 
+    void saveContent(byte[] content, Path fileName, String contentType);
```

Comment:
I think all 3 `saveContent` calls could/should flow top-to-bottom into a single one that does the work.  Right now it's a little split and hard to follow.